core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2448,7 +2449,7 @@ private[spark] object Utils
     } catch {
       // JEP 486: Permanently Disable the Security Manager
       case e: UnsupportedOperationException if e.getMessage().contains("getSubject") =>
-        "spark"
+        System.getProperty("user.name", "spark")
```

Comment:
Thank you @dongjoon-hyun , BTW, is it possible to use `anonymous` instead of `spark`? AFAIK, many Hadoop/Spark distributions tend to use `spark` as the admin user for the Spark component.

Reply:
This is the same case when you use `SPARK_USER=spark`.

Security is already broken here when `getSubject` is disallowed, @yaooqinn .

Reply:
Maybe, do you consider this as alternative of `ACL`?

Reply:
I found that we can follow this. WDYT, @yaooqinn ?
https://github.com/apache/spark/blob/55c65d982460b5de3e2dcec64d45e74b7df94277/core/src/main/scala/org/apache/spark/deploy/ApplicationDescription.scala#L36

Reply:
I can find `185` user id for Docker and K8s, but I'm not aware of `anoymous` in Spark code.
```
$ git grep user.name | grep anonymous | grep -v event
```

Reply:
Yes, @dongjoon-hyun, I was considering ACL kinds of stuff when running spark in client/server mode like apache/kyuubi. 

> System.getProperty("user.name", "<unknown>")

LGTM to the above